dpkg-diffs

```diff
@@ -118,43 +119,49 @@ function getCachedDebFile {
    pkgNameSpec=${1%=*}
    version=${1#*=}
    candidates="$APT_ARCHIVE/${pkgNameSpec}_${version}_*.deb"
-   if [ -z "$version" -o "$version" = "$1" ]; then
+   if [[ -z "$version" || "$version" = "$1" ]]; then
       if version_arch=$(dpkg-query -W -f '${Version}_${Architecture}' "$pkgNameSpec" 2>/dev/null); then
-         foundFile="$APT_ARCHIVE/${pkgNameSpec}_${version_arch}.deb"
+         # Replace ":" (the epoch separator) with "%3a" to match the file name.
+         foundFile="$APT_ARCHIVE/${pkgNameSpec}_${version_arch/:/%3a}.deb"
       else
          candidates="$APT_ARCHIVE/${pkgNameSpec}_*.deb"
       fi
    fi
    if [ -z "$foundFile" ]; then
-      if echo $candidates | grep -q ' '; then
+      if echo "$candidates" | grep -q ' '; then
          echo "Found more than one file in $APT_ARCHIVE matching \"$1\":" >&2
-         for i in $candidates; do 
+         for i in $candidates; do
             echo " ${i##*/}" >&2
          done
-	 die "Please provide a more precise package specification."
+     die "Please provide a more precise package specification."
       else
-         foundFile=$(echo $candidates)
+         foundFile="$candidates"
       fi
    fi
-   if [ ! -e $foundFile ]; then
+   if [ ! -e "$foundFile" ]; then
       echo "No package matching \"$1\" found in cache." >&2
```

Comment:
It would be helpful to include the following for those who don't read docs: `Use -d to download.`